recipes/onnx/all/conanfile.py

```diff
@@ -95,6 +97,7 @@ def generate(self):
         tc.variables["ONNX_VERIFY_PROTO3"] = Version(self.dependencies.host["protobuf"].ref.version).major == "3"
         if is_msvc(self):
             tc.variables["ONNX_USE_MSVC_STATIC_RUNTIME"] = is_msvc_static_runtime(self)
+        tc.variables["ONNX_DISABLE_STATIC_REGISTRATION"] = self.options.disable_static_registration
```

Comment:
the option is available only since 1.9 version.
I suppose you need to handle this option additionally in `config_options` method